backend/lib/piece-indexer.js

```diff
@@ -0,0 +1,82 @@
+import { fetchPayloadCid } from './pix-service/service.js'
+import { parseDeals } from './deal-observer.js'
+import assert from 'node:assert'
+import * as util from 'node:util'
+
+/** @import {Queryable} from '@filecoin-station/deal-observer-db' */
+/** @import { Static } from '@sinclair/typebox' */
+/** @import { ActiveDealDbEntry } from '@filecoin-station/deal-observer-db/lib/types.js' */
+
+/**
+ * @param {Queryable} pgPool
+ * @param {function} makeRpcRequest
+ * @param {function} makePixRequest
+ * @returns {Promise<void>}
+ */
+export async function updatePayloadCids (pgPool, makeRpcRequest, activeDeals, makePixRequest) {
+  const updatedDeals = []
+  for (const deal of activeDeals) {
+    const payloadCid = await fetchPayloadCid(deal.miner_id, deal.piece_cid, makeRpcRequest, makePixRequest)
+    deal.payload_cid = payloadCid
+    updatedDeals.push(deal)
+  }
+  await updatePayloadInActiveDeal(pgPool, updatedDeals)
```

Comment:
This is not great:

- We loop over all deals to update the in-memory deal objects with PayloadCID.
- Then inside `updatePayloadInActiveDeals`, we loop through those deals again to run one SQL query per deal.

Wouldn't it be better to have only one loop?

```js
  const updatedDeals = []
  for (const deal of activeDeals) {
    const payloadCid = await fetchPayloadCid(deal.miner_id, deal.piece_cid, makeRpcRequest, makePixRequest)
    deal.payload_cid = payloadCid
    await updatePayloadInActiveDeal(pgPool, updatedDeals)
  }
```

This is especially important in case `fetchPayloadCid` throws an error after we already resolved some payload CIDs. The current implementation will discard (not persist) the already resolved payload CIDs.